packages/msw-addon/README.md

```diff
@@ -65,3 +65,69 @@ SuccessBehavior.parameters = {
 ```
 
 The msw parameter takes an array of handlers as shown in [MSW official guide](https://mswjs.io/docs/getting-started/mocks/rest-api).
+
+The msw parameter can also be an object with named handlers or named arrays of handlers, to inherit (and optionally overwrite/disable) handlers from preview.js.
+
+```js
+
+//preview.js
+export const parameters = {
+    msw: {
+        auth: [
+            rest.get('/login', (req, res, ctx) => {
+                return res(
+                    ctx.json({
+                        success: true,
+                    })
+                )
+            }),
+            rest.get('/logout', (req, res, ctx) => {
+                return res(
+                    ctx.json({
+                        success: true,
+                    })
+                )
+            }),
+        ],
+    }
+};
+
+
+// story will include the auth handlers from preview.js
+SuccessBehavior.parameters = {
+  msw: {
+      profile: rest.get('/profile', (req, res, ctx) => {
+          return res(
+              ctx.json({
+                  firstName: 'Neil',
+                  lastName: 'Maverick',
+              })
+          )
+      }),
+  }
+}
+
+// story will overwrite the auth handlers from preview.js
+FailureBehavior.parameters = {
+    msw: {
+        auth: rest.get('/login', (req, res, ctx) => {
+            return res(ctx.status(403))
+        }),
+    }
+}
+
+// story will disable the auth handlers from preview.js
+NoAuthBehavior.parameters = {
+    msw: {
+        auth: null,
+        others: [
```

Comment:
I'm afraid we don't aim to support a map of handlers as it's not the most suitable format for what handlers represent.

`msw` parameter in Storybook must be a _list_ of handlers. You can compose those lists much easier than you can compose objects:

```js
import { auth, user, checkout, googleMaps } from './handlers'

SoryOne.parameters = {
  msw: [auth,  user]
}

StoryTwo.parameters = {
  msw: [auth, checkout, user, googleMaps]
}
```

Request handlers are not exclusive by design, but rather accumulative. You can have multiple handlers for the same endpoint and utilize their declaration order to implement more complex scenarios. Keeping handlers in a map implies the key-value uniqueness that'd be specific to this add-on only.

Reply:
Storybook internally calls a function `combineParameters` which recursively merges the parameters from:
preview.js -> story meta -> story. 
It is true that by accepting an array rather than an object at the main level of the parameters makes this addon lose the parameter inheritance, but that's not necessarily bad in the scenario for this addon. It really comes down to what we're looking for @kettanaito. Maybe in the future there will be other properties for this addon, and it might be a good idea to turn it into an object already. Most addons (all of @storybook/essentials) use parameters as objects, for instance.

There's another PR (#12) that was trying to change the way the handlers are passed as well. I guess we need to think what problems we are trying to solve and come up with a good solution for them all.

In the meantime, I think @kettanaito's suggestion is pretty solid, even though you will need to always import the handler to use it.

You could always create a helper function like:
```ts
// handlers.js
const globalHandlers = [...]
export const composeMswHandlers = (handlers) => [...globalHandlers, ...handlers]
```

Reply:
I've replied on the [issue](https://github.com/mswjs/msw-storybook-addon/issues/43#issuecomment-892240965) I created regarding this PR.